### PR TITLE
don't allow dragging TCOs in BBEditor

### PR DESCRIPTION
**Pull request by BaraMGB**
fixes https://github.com/LMMS/lmms/issues/3822

### CODE REVIEW

**Comment by lukas-w** [comment]
For better modularity: Can you try implementing the fix in the affected subclass (`BBTCOView`?) instead of using `dynamic_cast` in the super class? I.e. override `mousePressEvent` in `BBTCOView` and stop the events from being propagated there.

**Comment by BaraMGB** [comment]
I'm not sure. I looked into this. But a BBTCOView is this one:

bildschirmfoto von 2017-09-28 14-29-45

The issue appears with all TCOs in a BBTrackContainer.

**Comment by tresf** [comment]
I was under the impression [`fixedTCOs`](https://github.com/LMMS/lmms/blob/32f07ac317b42dda42a1d00b0a3b1fa1feef0655/include/BBEditor.h#L75) was the boolean flag that stored whether or not a pattern was resizable.

**Comment by lukas-w** [comment]
Didn't know about `fixedTCOs`, looks like this is what we need here. 👍

**Comment by BaraMGB** [comment]
Sometimes it's easier than imagined. 🙄

File: src/core/Track.cpp [hunk 1]
```diff
@@ -688,9 +688,7 @@ void TrackContentObjectView::paintTextLabel(QString const & text, QPainter & pai
 void TrackContentObjectView::mousePressEvent( QMouseEvent * me )
 {
 	setInitialMousePos( me->pos() );
-
-	BBTrackContainer * isbbEditor = dynamic_cast<BBTrackContainer*>( getTrackContentObject()->getTrack()->trackContainer() );
-	if( !isbbEditor && me->button() == Qt::LeftButton )
+	if( me->button() == Qt::LeftButton && fixedTCOs() == false )
```
Comment (Umcaruje) [inline src/core/Track.cpp]: You could just to `!FixedTCOs()` since it's a boolean

**Comment by Umcaruje** [comment]
Just tested this out, works well, fixes the bug :+1:

File: src/core/Track.cpp (hunk 1 above)
Comment (BaraMGB) [reply src/core/Track.cpp]: I wanted to be consist with the rest of the code. But if there are no objections I do it for the whole If statement.

Comment (Umcaruje) [reply src/core/Track.cpp]: `!FixedTCOs()` because I literally read it "If it's not a Fixed TCO"

**Pull request merged**